dan_layer/template_lib/src/resource/manager.rs

```diff
@@ -109,6 +109,22 @@ impl ResourceManager {
         })
     }
 
+    pub fn mint_many_non_fungible<T: Encode, U: Encode>(
+        &mut self,
+        metadata: &T,
+        mutable_data: &U,
+        supply: usize,
+    ) -> Bucket {
+        let mut tokens: HashMap<NonFungibleId, (Vec<u8>, Vec<u8>)> = HashMap::with_capacity(supply);
+        for _ in 0..supply {
+            let id = NonFungibleId::random();
+            tokens.insert(id, (encode(metadata).unwrap(), encode(mutable_data).unwrap()));
```

Comment:
nit(perf): data and mutable_data never change so you can encode them once and reuse in the loop

```suggestion
            tokens.insert(id, (encode(metadata).unwrap(), encode(mutable_data).unwrap()));
```